templates/test/cpp/naming/resolver_component_tests_defs.include

```diff
@@ -41,10 +43,10 @@ echo "Local DNS server started. PID: $DNS_SERVER_PID"
 # Health check local DNS server TCP and UDP ports
 for ((i=0;i<30;i++));
 do
-  echo "Retry health-check DNS query to local DNS server over tcp and udp"
+  echo "Retry health-check local DNS server by attempting a DNS query and TCP handshake"
   RETRY=0
-  dig A health-check-local-dns-server-is-alive.resolver-tests.grpctestingexp. @localhost -p "$FLAGS_test_dns_server_port" +tries=1 +timeout=1 | grep '123.123.123.123' || RETRY=1
-  dig A health-check-local-dns-server-is-alive.resolver-tests.grpctestingexp. @localhost -p "$FLAGS_test_dns_server_port" +tries=1 +timeout=1 +tcp | grep '123.123.123.123' || RETRY=1
+  $FLAGS_test_dns_resolver_bin_path -s 127.0.0.1 -p "$FLAGS_test_dns_server_port" -n health-check-local-dns-server-is-alive.resolver-tests.grpctestingexp. -t 1 | grep '123.123.123.123' || RETRY=1
```

Comment:
any ipv6 concerns? 127.0.0.1 is ipv4-specific. @pmarks-net

Reply:
Note the test is already broken under ipv6-only as is (https://github.com/grpc/grpc/blob/master/test/cpp/naming/resolver_component_tests_runner.sh#L168)

Reply:
Yes, `127.0.0.1` is incompatible with our `ipv6_only` test environment; gRPC is a http://cs/broken_under_ipv6_only bloodbath, but I don't recommend making the problem worse.

Reply:
As I understand, the problem is that the DNS server which this script (`$FLAGS_test_dns_resolver_bin_path`) is querying is currently binding to the ipv4 `INADDR_ANY` address ([this python library call to start listening on a port](https://github.com/grpc/grpc/blob/master/test/cpp/naming/test_dns_server.py#L103) uses that by default, [configured around here](https://github.com/twisted/twisted/blob/trunk/src/twisted/internet/tcp.py#L973).

So this `$FLAGS_test_dns_resolver_bin_path` script I believe needs to set it's target address to `127.0.0.1` to be sure to reach it. The same problem exists for the gRPC [test binary](https://github.com/grpc/grpc/blob/master/test/cpp/naming/resolver_component_tests_runner.sh#L168). Neither of these clients can resolve `localhost` too, so I think that fixing this `broken-under ipv6-only` situation for this test would require conditionally checking ipv6 availability for both of the clients and the server.

Would we be ok filing a bug about this test's ipv6-only brokenness and fixing in a separate PR?

Reply:
Filed https://github.com/grpc/grpc/issues/14644 for the ipv6 brokenness